### DISCUSSION TITLE
Sending invitations from within a flow

### DISCUSSION THREAD

**Discussion by k0l0ssus** [directus/directus#16220] [Archive]
Hi: 

I'm using Flows to invoke the invite API. The invite is triggered on creation of an Invite entity, which I'm using to control the invitation process.  From the Flow log:

```{
  "method": "POST",
  "url": "http://13.233.161.172:443/users/invite",
  "body": "{\"email\": \"fakemcfakerson@mail.com\",\n  \"role\" : \"0e18b6c9-074c-44e1-azb7-04c35be8ee17\"}",
  "headers": null
}
```

The response:

```{
  "message": "Request failed with status code 400",
  "name": "AxiosError",
  "config": {
    "transitional": {
      "silentJSONParsing": true,
      "forcedJSONParsing": true,
      "clarifyTimeoutError": false
    },
    "transformRequest": [
      null
    ],
    "transformResponse": [
      null
    ],
    "timeout": 0,
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
    "maxContentLength": -1,
    "maxBodyLength": -1,
    "env": {},
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Content-Type": "application/x-www-form-urlencoded",
      "User-Agent": "axios/0.27.2",
      "Content-Length": 85
    },
    "url": "http://13.233.161.172:443/users/invite",
    "method": "post",
    "data": "{\"email\": \"fakemcfakerson@mail.com\",\n  \"role\" : \"0e18b6c9-074c-44e1-azb7-04c35be8ee17\"}",
  },
  "code": "ERR_BAD_REQUEST",
  "status": 400
}
```

With debug level logging on, I see the 400 reported as:

```
06:50:36 🐛 "email" is required
06:50:36 ✨ request completed POST 400 /users/invite 19ms
```
I'm very clearly sending the email field, and I have SES configured in full. What "email" field am I missing still?

Dockerized Directus v 9.13, deployed in ECS. Thank you

**Comment by azrikahar** [comment]
It seems like the data is sent as _stringified_ json: `"{\"email\": \"fakemcfakerson@mail.com\",\n  \"role\" : \"0e18b6c9-074c-44e1-azb7-04c35be8ee17\"}"`, but this should be fixed by #15946 in `9.19.0` and above 👍

> I'm very clearly sending the email field, and I have SES configured in full. What "email" field am I missing still?

Since it is a stringified JSON, and the resulting Content-Type header sent ended up being `application/x-www-form-urlencoded` rather than `application/json`, the API technically only receives an empty _json_ body, which is why it is interpreting it as `email` was never sent. Hope that clears things up on this aspect!

**Reply by k0l0ssus** [reply]
Brilliant, thank you
